pkg/retriever/retriever_test.go

```diff
@@ -301,9 +357,9 @@ func TestRetriever(t *testing.T) {
 			}
 
 			// --- create ---
-			ret, err := retriever.NewRetriever(context.Background(), config, client, candidateFinder, dummyBlockConfirmer)
+			ret, err := retriever.NewRetriever(context.Background(), config, client, candidateFinder)
 			require.NoError(t, err)
-			ret.RegisterListener(eventsListener)
+			ret.RegisterSubscriber(subscriber.Collect)
 
 			// --- start ---
 			select {
```

Comment:
again, is there a concrete case where the retriever doesn't start? It just appears to be calling eventmanager.Start() which doesn't appear to have a case where it doesn't start really.